src/main/java/io/reactivex/schedulers/TestScheduler.java

```diff
@@ -23,9 +23,60 @@
 import io.reactivex.internal.functions.ObjectHelper;
 
 /**
- * A special, non thread-safe scheduler for testing operators that require
- * a scheduler without introducing real concurrency and allows manually advancing
- * a virtual time.
+ * This special {@link Scheduler} allows testing asynchronous and/or timed operators
+ * and flows by programatically executing the scheduled tasks up to a certain virtual
+ * time, allowing checking for intermediate results, state changes or preparing the relevant
+ * context (global/captured object state) before executing the next set of tasks.
+ * <p><p>
+ * Tasks scheduled via {@link #scheduleDirect(Runnable)} or
+ * {@link Scheduler.Worker#schedule(Runnable)},
+ * with or without delay, won't be executed until the virtual time is moved forward
+ * via the {@link #advanceTimeBy} or {@link #advanceTimeTo} methods of this
+ * {@code TestScheduler}. Tasks scheduled without delay can be executed without
+ * moving the virtual time forward via the {@link #triggerActions} method.
+ * Note that these methods are not thread safe and only one thread should is
```

Comment:
"should is allowed" doesn't seem to be grammatically correct

Reply:
Fixed.